client/state/plugins/installed/selectors.js

```diff
@@ -105,6 +105,16 @@ export function getPlugins( state, siteIds, pluginFilter ) {
 	return sortBy( pluginList, ( item ) => item.slug.toLowerCase() );
 }
 
+export const getPlugins = createSelector(
+	getPluginsSelector,
+	( state ) => [ state.plugins.installed.plugins ],
+	( state, siteIds, pluginFilter ) => {
+		const pluginIds = Object.keys( state.plugins.installed.plugins );
+
+		return [ siteIds, pluginIds, pluginFilter ].join();
```

Comment:
Do we take for granted that the data of each plugin will itself not change? 

I think that's a false assertion because a plugin for a site might get changed, and the selector must react to that change. Examples of such a change are:

* Toggling autoupdate for a plugin of a site.
* Activating/deactivating a plugin for a site.
* Updating a plugin for a site.

That means that tracking only the active plugin slugs won't be enough to keep a fresh list of plugins.

Reply:
In the case that data on an individual plugin changes the selector will re-compute because the redux state changes (i.e. the dependency function above gets it). I'm actually not sure why I had the plugin IDs in there, they are probably a leftover from experiments I was doing, so I took them out.